File: 905_Sort_Array_By_Parity/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortArrayByParity(int* nums, int numsSize, int* returnSize) {
    int* result = (int*)malloc(numsSize * sizeof(int));
    int even = 0, odd = numsSize - 1;

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] % 2 == 0) {
            result[even] = nums[i];
            even++;
        } else {
            result[odd] = nums[i];
            odd--;
        }
    }

    *returnSize = numsSize;
    return result;
}

File: 905_Sort_Array_By_Parity/solution.py
# class Solution:
#     def sortArrayByParity(self, nums: List[int]) -> List[int]:
#         even = []
#         odd = []
#         for num in nums:
#             if num % 2 == 0:
#                 even.append(num)
#             else:
#                 odd.append(num)

#         return even + odd

class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        n = len(nums)
        even, odd = 0, n - 1
        result = [0] * n

        for i in range(n):
            if nums[i] % 2 == 0:
                result[even] = nums[i]
                even += 1
            else:
                result[odd] = nums[i]
                odd -= 1

        return result